aws/aws.go

```diff
@@ -56,3 +60,43 @@ func (co ConfigOverrider) ClientConfig(serviceName string, cfgs ...*aws.Config)
 	cfgs = append(co.Configs[:len(co.Configs):len(co.Configs)], cfgs...)
 	return co.Base.ClientConfig(serviceName, cfgs...)
 }
+
+// ConfigFromURLParams returns an aws.Config initialized based on the URL
+// parameters in q. It is intended to be used by URLOpeners for AWS services.
+//
+// It returns an error if q contains any unknown query parameters; callers
+// should remove any query parameters they know about from q before calling
+// ConfigFromURLParams.
+//
+// The following query options are supported:
+//  - region: The AWS region for requests; sets aws.Config.Region.
+//  - endpoint: The endpoint URL (hostname only or fully qualified URI); sets aws.Config.Endpoint.
+//  - disableSSL: A value of "true" disables SSL when sending requests; sets aws.Config.DisableSSL.
+//  - s3ForcePathStyle: A value of "true" forces the request to use path-style addressing; sets aws.Config.S3ForcePathStyle.
```

Comment:
Optional: this option is only applicable to S3 requests, perhaps make it a blob-specific parameter?

Reply:
Interesting. Again, this seems not ideal by Amazon. But, I think I'd rather leave all of the `aws.Config` stuff together rather than try to "fix" it for them.